### DISCUSSION TITLE
Conflict between dynamic routes and static files

### DISCUSSION THREAD

**Discussion by tmatek** [vercel/next.js#13626] [Help]
Greetings,

we are developing a NextJS app that will replace part of existing website. The app will be mounted onto a sub-path at `/user/subpath`. We want to support URLs such as `/user/subpath/[urlKey]`, so in the pages folder we created a `[slug].js` file. The problem is that the [slug].js file now matches urls like `/favicon.ico` and `/2x` (in development). Is there any way to exclude those files, by somehow prefixing static assets?

**Comment by ijjk** [comment]
Hi, if the `favicon.ico` file exists in your `public` folder it will be matched before the dynamic route. You can also add a [pages/404.js](https://nextjs.org/docs/advanced-features/custom-error-page#404-page) file and leverage the new experimental [custom routes](https://github.com/vercel/next.js/issues/9081) feature to point these files to the 404 page directly if you would like although note the rewrite will only apply if the files do not exist in the `public` folder since the public folder has priority over the rewrites

```
module.exports = {
  experimental: {
    async rewrites() {
      return [
        {
          source: '/favicon.ico',
          destination: '/404' // or some other page/public file
        }
      ]
    }
  }
}
```

**Comment by 1aurabrown** [comment]
I'm having a similar experience with robots.txt getting picked up by [slug].js—it doesn't seem to be the case that files in /public get checked before dynamic routes.
